src/Arcus.WebApi.Correlation/IServiceCollectionExtensions.cs

```diff
@@ -33,7 +33,7 @@ public static IServiceCollection AddCorrelation(this IServiceCollection services
             Guard.NotNull(services, nameof(services));
 
             services.AddHttpContextAccessor();
-            services.AddSingleton<HttpCorrelationInfo>();
+            services.AddTransient<HttpCorrelationInfo>();
```

Comment:
Transient or [scoped](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-3.1#scoped)?

Reply:
I think transient is more safe/stable in this context, no?

Reply:
Not really because the correlation info will be re-created every time instead of on a per-request basis, unless I'm missing something?

Reply:
Hmm, the tests fails if I change it to scoped. The log properties are in that case a little behind.

Reply:
Hm I'll trust the tests then, but it's strange :D

Reply:
We could look into it deeper if you want, afterwards. Issue for this?

Reply:
I get how it works now, I think it's ok for now